src/ray_julia_jll/common.jl

```diff
@@ -157,21 +157,17 @@ let msg_types = (Address, JobConfig, ObjectReference)
     for T in msg_types
         types = (Symbol(nameof(T), :Allocated), Symbol(nameof(T), :Dereferenced))
         for A in types, B in types
-            @eval function Base.:(==)(a::$A, b::$B)
-                serialized_a = safe_convert(String, SerializeAsString(a))
-                serialized_b = safe_convert(String, SerializeAsString(b))
-                return serialized_a == serialized_b
-            end
+            @eval Base.:(==)(a::$A, b::$B) = SerializeAsString(a) == SerializeAsString(b)
         end
     end
 end
 
 function Serialization.serialize(s::AbstractSerializer, message::Message)
-    serialized_message = safe_convert(String, SerializeAsString(message))
+    serialized_message = SerializeAsString(message)
 
     serialize_type(s, Message)
     serialize(s, supertype(typeof(message)))
-    serialize(s, serialized_message)
+    serialize(s, String(serialized_message))
```

Comment:
do we need the `String` here?

Reply:
Yes, serializing a `StdString` doesn't work at the moment. See: https://github.com/JuliaInterop/CxxWrap.jl/pull/381#issuecomment-1773402995